test/spec/functions/mergePure.js

```diff
@@ -0,0 +1,240 @@
+'use strict';
+const merge = require('../../../src/functions/mergePure');
+
+it('should merge `source` into `object`', function() {
+  var names = {
+    characters: [{name: 'barney'}, {name: 'fred'}]
+  };
+
+  var ages = {
+    characters: [{age: 36}, {age: 40}]
+  };
+
+  var heights = {
+    characters: [{height: '5\'4"'}, {height: '5\'5"'}]
+  };
+
+  var expected = {
+    characters: [
+      {name: 'barney', age: 36, height: '5\'4"'},
+      {name: 'fred', age: 40, height: '5\'5"'}
+    ]
+  };
+
+  expect(merge(names, ages, heights)).toEqual(expected);
+});
+
+it('should merge sources containing circular references', function() {
+  var object = {
+    foo: {a: 1},
+    bar: {a: 2}
+  };
+
+  var source = {
+    foo: {b: {c: {d: {}}}},
+    bar: {}
+  };
+
+  source.foo.b.c.d = source;
+  source.bar.b = source.foo.b;
+
+  var actual = merge(object, source);
+
+  expect(actual.bar.b).not.toEqual(actual.foo.b);
+  expect(actual.foo.b.c.d).toEqual(actual.foo.b.c.d.foo.b.c.d);
+});
+
+it('should work with four arguments', function() {
+  var expected = {a: 4};
+  var actual = merge({a: 1}, {a: 2}, {a: 3}, expected);
+
+  expect(actual).toEqual(expected);
+});
+
+it('should merge onto function `object` values', function() {
+  function Foo() {}
+
+  var source = {a: 1};
+  var actual = merge(Foo, source);
+
+  expect(actual).toEqual(Foo);
+  expect(Foo.a).toEqual(1);
+});
+
+it('should merge first source object properties to function', function() {
+  var fn = function() {};
+  var object = {prop: {}};
+  var actual = merge({prop: fn}, object);
+
+  expect(actual.prop).toBeInstanceOf(Function);
```

Comment:
Can you explain to me why actual.prop should be a function instead of {} ?

Reply:
if the source is a function, we want it to stay as a function. Not too important for our use case, but I didn't beak it

Reply:
I see. then why does [this](https://github.com/algolia/algoliasearch-helper-js/pull/694/commits/4f1ab4d5e4a23b7489525706332c384b794e5c25#diff-e96e00593d1b2b63d3b8689232becc30R77) replace a function with {} ? It's a bit confusing to me.

Reply:
that was the original test from lodash, it also didn't make sense to me, so I changed it: https://github.com/algolia/algoliasearch-helper-js/pull/694/files#diff-9de5a55358378fba3c54e69c20f364f9R79

Reply:
thanks for the clarification. I was reading from "commits" tab and apparently it wasn't showing me the latest force-pushed commits.